Bio/PDB/ccealignmodule.c

```diff
@@ -211,6 +211,149 @@ calcS(
+    5.19, 5.33, 5.46, 5.60, 5.73, 5.86, 5.99, 6.11, 6.23, 6.35,
+    6.47, 6.58, 6.70, 6.81, 6.92, 7.02, 7.13, 7.24, 7.34, 7.44,
+    7.54, 7.64, 7.74, 7.84, 7.93, 8.03, 8.12, 8.21, 8.30, 8.40,
+    8.49, 8.57, 8.66, 8.75, 8.84, 8.92, 9.01, 9.09, 9.17, 9.25,
+    9.34, 9.42, 9.50, 9.58, 9.66, 9.73, 9.81, 9.89, 9.97, 10.04,
+    10.12, 10.19, 10.27, 10.34, 10.41, 10.49, 10.56, 10.63, 10.70, 10.77,
+    10.84, 10.91, 10.98, 11.05, 11.12, 11.19, 11.26, 11.32, 11.39, 11.46,
+    11.52, 11.59, 11.66, 11.72, 11.79, 11.85, 11.91, 11.98, 12.04, 12.10,
+    12.17, 12.23, 12.29, 12.35, 12.42, 12.48, 12.54, 12.60, 12.66, 12.72,
+    12.78, 12.84, 12.90, 12.96, 13.02, 13.07, 13.13, 13.19, 13.25, 13.31,
+    13.36, 13.42, 13.48, 13.53, 13.59, 13.65, 13.70, 13.76, 13.81, 13.87,
+    13.92, 13.98, 14.03, 14.09, 14.14, 14.19, 14.25, 14.30, 14.35, 14.41,
+    14.46, 14.51, 14.57, 14.62, 14.67, 14.72, 14.77, 14.83, 14.88, 14.93};
+
+static double zToP(const double z)
+{
+    int index = (int)(z / 0.1);
+
+    if (index < 0) {
+        index = 0;
+    }
+    if (index > 149) {
+        index = 149;
+    }
+
+    return tableZtoP[index];
+}
+
+static double pToZ(const double p)
+{
+    int index = (int)(-log10(p) * 3.0);
+
+    if (index < 0) {
+        index = 0;
+    }
+    if (index > 149) {
+        index = 149;
+    }
+
+    return tablePtoZ[index];
+}
+
+static const double similarityAvgs[] =
+    {2.54, 2.51, 2.72, 3.01, 3.31, 3.61, 3.90, 4.19, 4.47, 4.74,
+        4.99, 5.22, 5.46, 5.70, 5.94, 6.13, 6.36, 6.52, 6.68, 6.91};
+static const double similaritySDs[] =
+    {1.33, 0.88, 0.73, 0.71, 0.74, 0.80, 0.86, 0.92, 0.98, 1.04,
+        1.08, 1.10, 1.15, 1.19, 1.23, 1.25, 1.32, 1.34, 1.36, 1.45};
+
+static double zScoreSimilarity(
+    const int fragmentSize,
+    const int pathLength,
+    const double similarity)
+{
+    if (fragmentSize != 8 || pathLength < 1) {
+        return 0.0;
+    }
+
+    double similarityAvg, similaritySD;
+
+    if (pathLength < 21) {
+        similarityAvg = similarityAvgs[pathLength - 1];
+        similaritySD = similaritySDs[pathLength - 1];
+    }
+    else {
+        similarityAvg = 0.209874 * pathLength + 2.944714;
+        similaritySD = 0.039487 * pathLength + 0.675735;
+    }
+    if (similarity > similarityAvg) {
+        return 0.0;
+    }
+
+    return (similarityAvg - similarity) / similaritySD;
+}
+
+static const double gapCountAvgs[] =
+    {0.00, 11.50, 23.32, 35.95, 49.02, 62.44, 76.28, 90.26,
+        104.86, 119.97, 134.86, 150.54, 164.86, 179.57, 194.39,
+        209.38, 224.74, 238.96, 253.72, 270.79};
+static const double gapCountSDs[] =
+    {0.00, 9.88, 14.34, 17.99, 21.10, 23.89, 26.55, 29.00, 31.11,
+        33.10, 35.02, 36.03, 37.19, 38.82, 41.04, 43.35, 45.45,
+        48.41, 50.87, 52.27};
+
+static double zScoreGapCount(
+    const int fragmentSize,
+    const int pathLength,
+    const int gapCount)
+{
+    if (fragmentSize != 8 || pathLength < 2) {
```

Comment:
Why `pathLength < 2` here but above was `< 1`?

Reply:
That is how the [original source code](https://github.com/kad-ecoli/CE/blob/master/ce/ce_align.C#L1053) is. I did notice that the first element in the `gapCountAvgs` array is `0.00`. In the case where `pathLength` is `1`, the function still returns zero if the condition is changed to `pathLength < 1`. So i will go ahead and change it for consistency.